### ISSUE TITLE
Report issue button functionality

### ISSUE THREAD

**Issue by TechSupport26** [holochain/launcher-tauri#203]
**Describe the bug**
I would assume that the Report Bug on GitHub button would direct you to the appropriate page however it does not do this.
Also there is a UI issue shown in the video provided where the popups are not always in the correct location.

**To Reproduce**

1. Open the launcher and sign in.
2. Click on the bug report button.
3. The GitHub Page should open.

**Expected behavior**
The https://github.com/holochain/launcher/issues/new page should open

**Logs**
N/A

> In Ubuntu, there is a weird error that prevents the upload of a file from inside a hidden folder. To work around it, you can copy the `launcher.log` file to another non-hidden location, and upload it from there.

**Screenshots**

The following shows the UI issue where the Settings and Report Bug on GitHub (https://github.com/holochain/launcher/issues/new) popup do not appear in the correct place and can be off the screen.

https://github.com/holochain/launcher/assets/65631811/42b29273-4d20-4a13-b716-ab7d5630bf81

**Desktop (please complete the following information):**

- OS: Debian 12 - Bookworm
- Holochain Launcher Version
Holochain Launcher v0.11.1 (Beta 0.2)

**Additional context**
N/A

**Comment by matthme** [comment]
Are you saying the "Report Bug" button does not do anything at all? It's working fine for me so I suspect it's got something to do with your linux distribution not supporting the command that is called to open the url in the default browser. What I will do as a next step is bump the version of the Rust crate that's being used and if we're lucky it has been fixed upstream already.

Does opening the logs folder work fine for you (Settings > Open Logs)? There we're using a different Rust crate to open the folder and I had reports that it's not working on Kubuntu. So it would be interesting to know whether that works fine on your Linux distribution.

**Comment by matthme** [comment]
And I think regarding the popups not displaying properly I think I cannot do much unfortunately since this is not something the Launcher UI actively controls. It's handled either by the operating system or the webview.

**Comment by TechSupport26** [comment]
> Are you saying the "Report Bug" button does not do anything at all

It changes the cursor (as if there was a button that you can click) but that is all.

> Does opening the logs folder work fine for you

It looks like it is trying. (see video) It appears to loose focus on the launcher as the bottom menu bar appeared so something is happening but not successfully.

https://github.com/holochain/launcher/assets/65631811/6aeda100-b022-4918-a740-8517dd857187

https://1drv.ms/u/s!AuXD18607r4Z11kntX2yVYKAVU4E?e=rxV4wJ
The log file is too big to upload here.
I tried to open the issues and the log file then close the launcher so it should be at the end of the file.

Let me know if it would be better to just delete the file and try both again so that there is less noise to sort through.

**Comment by TechSupport26** [comment]
> Let me know if it would be better to just delete the file and try both again so that there is less noise to sort through.

I did this anyway as the file takes too long to load.

[launcher.log](https://github.com/holochain/launcher/files/12825521/launcher.log)

It looks like there is nothing that indicates an error but I hope it helps.
Also I just realised that you never asked for the logs but perhaps in testing these actions can be recorded.

**Comment by matthme** [comment]
Thanks for the logfiles! As I feared, it did not even log an error in the log file so it's likely related to [this](https://github.com/tauri-apps/tauri/issues/6172) issue. And since both opening the logs and clicking the link doesn't work means that both available Rust crates (`opener` and `open`) don't seem to work on your OS. With the latest release (0.11.2) I bumped both of those crates to the latest version, you may try whether that helped (I'm not too confident though).
If not, it would be good to know whether `xdg-open` is installed on your system. You can figure it out by entering `xdg-open --version` on the command line.

**Comment by TechSupport26** [comment]
> Thanks for the logfiles! As I feared, it did not even log an error in the log file so it's likely related to [this](https://github.com/tauri-apps/tauri/issues/6172) issue. And since both opening the logs and clicking the link doesn't work means that both available Rust crates (`opener` and `open`) don't seem to work on your OS. With the latest release (0.11.2) I bumped both of those crates to the latest version, you may try whether that helped (I'm not too confident though).

I have tried with the latest version of the launcher and nothing works. There are some other holochain apps that allow me to open links so I would say there is another way to do it.

I have just installed the .dmg and I can open both the logs and the issues.
I don't know why the .appimage does not work.

Also I tried to run the appimage using sudo and it did not recognise the previous installation. Any ideas as to why? Also it did not help in any way that I could tell.

> If not, it would be good to know whether `xdg-open` is installed on your system. You can figure it out by entering `xdg-open --version` on the command line.

I looked at the comments and there were some other tools were installed.

```
xdg-open --version
xdg-open 1.1.3
gio --version
2.74.6
gnome-open --version
bash: gnome-open: command not found
kde-open --version
kioclient 5.27.5
wslview --version
bash: wslview: command not found
```

**Comment by TechSupport26** [comment]
Also should opening another website in the launcher be possible?

![image](https://github.com/holochain/launcher/assets/65631811/f22fe72f-1a9e-4bb0-a783-1e4686e9392d)

**Comment by matthme** [comment]
> There are some other holochain apps that allow me to open links so I would say there is another way to do it.

It's not really a Holochain problem but a tauri problem. Acorn for example uses [Electron](https://www.electronjs.org/) instead of tauri. I already try to catch clicks on links and the open them explicitly with the browser but apparently that's not working on your operating system :/ I'm not sure how to fix this if all available Rust crates seem to fail...I will have to think about it.

> I don't know why the .appimage does not work.

It works fine when I run the AppImage on my Ubuntu 22.04 so I think it's specific to your Linux distribution

> Also I tried to run the appimage using sudo and it did not recognise the previous installation. Any ideas as to why?

That's fascinating. I didn't know about that but I also see that when I run it on my Ubuntu 22.04. Maybe sudo stores things in different file locations - I need to look into it. Also, interstingly the menu items look differently when run with sudo so it seems to use a different version of webkit

> I looked at the comments and there were some other tools were installed.

Thanks! So if you have `xdg-open` installed it's probably really the same issue as the one I linked above since it's supposed to work then...

**Comment by matthme** [comment]
By the way, it's probably not a good idea to run AppImage files with `sudo` in general since this gives priviledges to an app that it normally should not have. So I don't think the launcher should should cover this case since it's not supposed to be run with `sudo`...

**Comment by TechSupport26** [comment]
> By the way, it's probably not a good idea to run AppImage files with `sudo` in general since this gives priviledges to an app that it normally should not have. So I don't think the launcher should should cover this case since it's not supposed to be run with `sudo`...

Yes. The idea behind it was that perhaps it was a permissions issue but it was not.

Just @ me if you need me to test anything. (I will probably see any comments anyway)
